lib/licensed/sources/gradle.rb

```diff
@@ -101,27 +108,50 @@ def enumerate_dependencies
       def gradle_executable
         return @gradle_executable if defined?(@gradle_executable)
         @gradle_executable = begin
-          gradlew = File.join(config.pwd, "gradlew")
-          return gradlew if File.executable?(gradlew)
-          "gradle" if Licensed::Shell.tool_available?("gradle")
-        end
+                               gradlew = File.join(config.pwd, "gradlew")
+                               return gradlew if File.executable?(gradlew)
+                               "gradle" if Licensed::Shell.tool_available?("gradle")
+                             end
       end
 
       # Returns the configurations to include in license generation.
       # Defaults to ["runtime", "runtimeClasspath"]
       def configurations
         @configurations ||= begin
-          if configurations = config.dig("gradle", "configurations")
-            Array(configurations)
-          else
-            DEFAULT_CONFIGURATIONS
-          end
+                              if configurations = config.dig("gradle", "configurations")
+                                Array(configurations)
+                              else
+                                DEFAULT_CONFIGURATIONS
+                              end
+                            end
+      end
+
+      def self.should_add_gradle_license_report_plugins_block?(gradle_build_file)
+        !/dependency-license-report/.match(gradle_build_file)
+      end
+
+      def self.build_gradle_contains_plugins_block?(gradle_build_file)
+        /plugins/.match(gradle_build_file)
+      end
+
+      def self.add_gradle_license_report_plugins_block?(gradle_build_file)
+        if(Licensed::Sources::Gradle.build_gradle_contains_plugins_block?(gradle_build_file))
+          gradle_build_file.gsub(/(?<=plugins)\s+{/, " { id 'com.github.jk1.dependency-license-report' version '1.6'")
+        else
+
+          gradle_build_file = " plugins { id 'com.github.jk1.dependency-license-report' version '1.6' }" + gradle_build_file
         end
       end
 
       def self.gradle_command(*args, path:, executable:, configurations:)
+        gradle_build_file = File.read("build.gradle")
+        if(Licensed::Sources::Gradle.should_add_gradle_license_report_plugins_block?(gradle_build_file))
```

Comment:
the new code is not as easy to read as it could be due to long and similar looking method names.  what do you think about inlining this check into `add_gradle_license_report_plugins_block` as a guard clause

```
def self.add_gradle_license_report_plugins_block(gradle_build_file)
   return unless gradle_build_file.include? "dependency-license-report"
   ...
```

then all the logic to ensure that the gradle file is 👍 is self contained in the new method